src/test/java/persistence/TestLessonPersistence.java

```diff
@@ -0,0 +1,90 @@
+package persistence;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static util.DatabaseOps.performQuery;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.List;
+import model.Lesson;
+import model.Student;
+import model.Subject;
+import model.Teacher;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.testcontainers.containers.PostgreSQLContainer;
+import org.testcontainers.utility.DockerImageName;
+
+public class TestLessonPersistence {
+  private static final PostgreSQLContainer databaseContainer =
+      new PostgreSQLContainer(DockerImageName.parse("postgres:alpine3.14"));
+  private LessonPersistence lessonPersistence;
+  private final Subject subjectPiano = new Subject(1L, "Piano", 60, List.of(1, 6), false);
+  private final Subject subjectGuitar = new Subject(2L, "Guitar", 45, List.of(1, 6), false);
+  private final Student studentAdam = new Student(1, "Adam", "Kowalski", 1);
+  private final Student studentJohn = new Student(2, "John", "Doe", 1);
+  private final Teacher teacherAdam =
+      new Teacher("uuid", "Adam", "Kowalski", "adam@sosnowa.pl", "razDwaTrzy");
+  private final Teacher teacherAndrzej =
+      new Teacher("id", "Andrzej", "Kowalski", "andrzej@sosnowa.pl", "razDwaCztery");
+
+  @BeforeEach
+  void initialize() throws Throwable {
+    databaseContainer.start();
+    String parentPath = Path.of(System.getProperty("user.dir")).getParent().toString();
+    Path path =
+        Path.of(parentPath + "/backend/src/main/resources/queries/database_initialization.sql");
+    performQuery(databaseContainer, Files.readString(path));
```

Comment:
ten kod jest powielany w 4 miejscach. Wyekstrahowałbym go do DatabaseOps
public static void runContainterAndIntializeDatabase(JdbcDatabaseContainer<?> container) {
        container.start();
        String parentPath = Path.of(System.getProperty("user.dir")).getParent().toString();
        Path path =
                Path.of(parentPath + "/backend/src/main/resources/queries/database_initialization.sql");
        try {
            performQuery(container, Files.readString(path));
        } catch (IOException | SQLException e) {
            e.printStackTrace();
        }
    }

Reply:
done